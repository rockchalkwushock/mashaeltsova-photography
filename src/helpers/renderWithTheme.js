import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
