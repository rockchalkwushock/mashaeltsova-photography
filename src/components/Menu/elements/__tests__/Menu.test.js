import React from 'react'

import { renderWithTheme } from '../../../../helpers'
import { Menu } from '../index'

describe('Styled Component: <Menu />', () => {
  let wrapper
  let wrapper2
  beforeAll(() => {
    wrapper = renderWithTheme(<Menu />)
    wrapper2 = renderWithTheme(<Menu state />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('background-color', '#333')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '0px 2px 5px 2px rgba(0,0,0,0.3)'
    )
    expect(wrapper).toHaveStyleRule('height', '60px')
    expect(wrapper).toHaveStyleRule(
      'transition',
      'height 0.3s ease-in-out 0.05s'
    )
    expect(wrapper).toHaveStyleRule('width', '100vw')
  })
  test('should have toggled styles', () => {
    expect(wrapper2).toHaveStyleRule('height', '160px')
  })
})
