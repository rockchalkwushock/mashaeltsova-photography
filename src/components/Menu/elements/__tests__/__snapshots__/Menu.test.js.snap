// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Component: <Menu /> should render markup 1`] = `
.c0 {
  background-color: #333;
  box-shadow: 0px 2px 5px 2px rgba(0,0,0,0.3);
  height: 60px;
  -webkit-transition: height 0.3s ease-in-out 0.05s;
  transition: height 0.3s ease-in-out 0.05s;
  width: 100vw;
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <Menu>
    <nav
      className="c0"
    />
  </Menu>
</ThemeProvider>
`;
